index.js

```diff
@@ -7,54 +7,76 @@ const path = require("path");
 app.use(express.static(path.join(__dirname, "public")));
 app.set("view engine", "ejs");
 
-// starting point coordinates
 const startLat = 43.839767;
 const startLng = 18.343170;
 
-// step distance
-const step = 0.01;
+const stepDistance = 100;
 
-// current point coordinates
 let currentLat = startLat;
 let currentLng = startLng;
 
-// grid of 25 points
+let prevLng = currentLng;
+
 const grid = [];
 for (let i = 0; i < 5; i++) {
     for (let j = 0; j < 5; j++) {
-        grid.push({
-            lat: startLat + i * step,
-            lng: startLng + j * step
-        });
+        if (j === 0) {
+            grid.push({
+                lat: move_coordinate_on_y_axis(currentLat, prevLng, i * stepDistance),
+                lng: prevLng
+            });
+        } else {
+            const lng = move_coordinate_on_x_axis(currentLat, currentLng, j * stepDistance);
+            grid.push({
+                lat: move_coordinate_on_y_axis(currentLat, lng, i * stepDistance),
+                lng: lng
+            });
+        }
     }
+   
+    prevLng = move_coordinate_on_x_axis(currentLat, currentLng, 5 * stepDistance);
+    currentLng = prevLng;
 }
 
-// current point index
-let currentIndex = 0;
-
-app.get("/uav/location", (req, res) => {
-    res.render("index", { grid, currentPoint: grid[currentIndex] });
+app.get("/uav-location", (req, res) => {
+    res.render("index", { grid, currentPoint: grid[0] });
 });
 
-io.on("connection", socket => {
+io.on("connection", async (socket) => {
     console.log("a user connected");
-    let gridIndex = 0;
-    let intervalId = setInterval(() => {
-        currentLat = grid[gridIndex].lat;
-        currentLng = grid[gridIndex].lng;
+
+    for (let i = 0; i < grid.length; i++) {
+        currentLat = grid[i].lat;
+        currentLng = grid[i].lng;
+
         socket.emit("updateCurrentPoint", { lat: currentLat, lng: currentLng });

```

Comment:
kada pripremis funkciju ```generatePlanePath```, onda ces da odradis fazon gdje ces da generises tri grida za tri razlicita aviona.

onda kada ih generises, onda unutar ovog for loopa, ces da ```socket.emit()``` te podatke.

npr:
```
let teamOne = generatePlanePath(1.23, 3.21, "teamOne", 5);
let teamTwo = ...;
let teamThree = ...;

socket.emit("...", [teamOne[i], teamTwo[i], teamThree[i]]);
```